specs/CookieManagement.md

```diff
@@ -0,0 +1,158 @@
+        Strict = 2,
+    };
+
+    runtimeclass CoreWebView2
+    {
+        /// Create a new cookie manager object.
+        CoreWebView2CookieManager CreateCookieManager();
+    }
+
+    /// Add or delete an CoreWebView2Cookie or CoreWebView2Cookies,
+    /// or view the cookies. The changes would apply to the context of the user profile.
+    /// That is, other WebViews under the same user profile could be affected.
+    runtimeclass CoreWebView2CookieManager
+    {
+        /// Create a cookie object with a specified name, value, domain, and path.
+        /// One can set other optional properties after cookie creation.
+        CoreWebView2Cookie CreateCookie(String name, String value, String Domain, String Path);
+
+        /// Gets a list of cookies matching the specific URI.
+        /// You can modify the cookie objects, call
+        /// CoreWebView2CookieManager.SetCookie, and the changes
+        /// will be applied to the webview.
+        Windows.Foundation.IAsyncOperation<CoreWebView2CookieList> GetCookiesAsync(String uri);
+
+        /// Sets a cookie with the given cookie data; may overwrite equivalent cookies
+        /// if they exist.
+        void SetCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes browser cookies with matching name and uri or domain/path pair.
+        /// Cookie name is required.
+        /// If uri is specified, deletes all cookies with the given name where domain
+        /// and path match provided URI.
+        /// If domain is specified, deletes only cookies with the exact domain.
+        /// If path is specified, deletes only cookies with the exact path.
+        void DeleteCookies(String name, String uri, String Domain, String Path);
+
+        /// Clears all cookies under the same profile.
+        /// This could affect other WebViews under the same user profile.
+        void ClearAllCookies();
+    }
+
+    /// A list of cookie objects.
+    runtimeclass CoreWebView2CookieList
+    {
+        /// The number of cookies contained in the CoreWebView2CookieList.
+        UInt32 Size { get; };
+
+        /// Get the cookie object at the given index.
+        CoreWebView2Cookie GetValueAtIndex(UInt32 index);
+    }
+
+    /// Provides a set of properties that are used to manage a CoreWebView2Cookie.
+    runtimeclass CoreWebView2Cookie
+    {
+        /// Cookie name.
+        String Name { get; };
+
+        /// Cookie value.
+        String Value { get; set; };
+
+        /// The domain for which the cookie is valid.
+        /// The default is the host that this cookie has been received from.
+        String Domain { get; };
+
+        /// The path for which the cookie is valid. If not specified, this cookie
+        /// will be sent to all pages on the Domain.
+        String Path { get; };
+
+        /// The expiration date and time for the cookie as the number of seconds since the UNIX epoch.

```

Comment:
A uint64 (or uint?) would be better than double for this. In COM we don't have a date/time interface so just the number seems OK, but in .NET and winrt it would be better to use the appropriate date/time classes from those languages/runtimes.

Reply:
As the Chromium cookie stores expires as a JavaScript number, we will use a double to represent the expires attribute here.